ui/app/appointments/services/calendarViewPopUp.js

```diff
@@ -47,6 +49,29 @@ angular.module('bahmni.appointments')
                     closeConfirmBox();
                 };
 
+                var isCurrentUserHavePrivilege = function (privilege) {
+                    return !_.isUndefined(_.find($rootScope.currentUser.privileges, function (userPrivilege) {
+                        return userPrivilege.name === privilege;
+                    }));
+                };
+
+                popUpScope.isUserAllowedToPerformEdit = function () {
```

Comment:
This also seems to be redefined. Would be better to have a way to reuse this code so that maintaining it will be easier.